doc/wg/core/notes/core-notes-2021-07-09.md

```diff
@@ -0,0 +1,157 @@
+* Hudson: Last comment on this is that we don't want to block 2.0 on stuff that hasn't been submitted or won't be all that quick if it's not ABI blocking. But, maybe we should look at the list of outstanding PRs and look for things to force through before release. My opinion is that we shouldn't do that this time, but maybe others disagree.
+* Branden: I agree with that strategy. 2.1 can be all the great updates to Tock, while 2.0 is the ABI changes and soundness issues.
+* Brad: I didn't see anything in particular I want to include.
+* Hudson: We can still merge things that are ready, of course.
+
+## Appslice swapping in capsules
+* Brad: Related question, I'm still confused about what we're promising to userspace around appslices and swapping. TRD says if you call allow you'll get the old buffer back. We're not enforcing that and leaving it as a TODO, which I think makes sense. But are we violating the TRD104 promises? Are we going to handle it soon? I think we need a record of where we stand.
+* Leon: I think this is a valid point and it's confusing because of how it developed. Hudson and I planned on enforcing restrictions before Tock 2.0, which is why it's written in the TRD104 this way. However, we've since decided we don't have to enforce for soundness and it's okay for now. But it would still be a good TODO. The TRD currently has expectations we don't meet in our codebase, so we should edit the TRD to say our expectation and our implementation.
+* Hudson: I thought we had said capsules were allowed to refuse to return a buffer. For example if we were passing it into a DMA request
+* Leon: Well, we can't use userspace buffers for DMA anyways. Capsules can refuse operations, for example if there is a complex piece of hardware. So if the buffer only goes away if the process dies, that makes things easier to handle. Otherwise _every_ operation would have to be abortable.
+* Hudson: I thought there was also the idea that a capsule can store info in an allowed buffer and could rely on it still being there. So we didn't want to let processes revoke buffers at any time.
+* Leon: Yes, but it can't be trusted, since the process could change data in the buffer whenever it wants.
+* Hudson: The reason I'm asking is that maybe the text in TRD104 is incorrect if we permit capsules to refuse an allow call. Which I thought was originally the plan.
+* Leon: I'm confused.
+* Hudson: Currently the documentation of TRD104, I believe, is that if userspace requests a buffer back, it must get it back. But that doesn't sync with my memory that allows can be refused.
+* Leon: I think the wording is that _if_ it gets a buffer back, it must be the right one. And the "must" restriction there is what we don't enforce.
+* Hudson: Okay. So we should submit a PR to TRD104 to modify the language a little and open an issue to enforce non swapping buffers.
```

Comment:
This is a good discussion to have. I think it reflects a changing perspective on the degree to which capsules are checked/validated and the Tock security model.